packages/flutter/test/material/bottom_sheet_theme_test.dart

```diff
@@ -132,6 +132,120 @@ void main() {
+      elevation: modalElevation,
+      persistentElevation: persistentElevation,
+    );
+    await tester.pumpWidget(bottomSheetWithElevations(bottomSheetTheme));
+    await tester.tap(find.byIcon(Icons.check));
+    await tester.pumpAndSettle();
+    final Material material = tester.widget<Material>(
+      find.descendant(
+        of: find.byType(BottomSheet),
+        matching: find.byType(Material),
+      ).first,
+    );
+    expect(material.elevation, persistentElevation);
+  });
+
+  testWidgets('BottomSheet persistentElevation takes priority over elevation - Modal', (WidgetTester tester) async {
+    const modalElevation = 5.0;
+    const persistentElevation = 7.0;
+    const bottomSheetTheme = BottomSheetThemeData(
+      elevation: modalElevation,
+      persistentElevation: persistentElevation,
+    );
+    await tester.pumpWidget(bottomSheetWithElevations(bottomSheetTheme));
+    await tester.tap(find.byIcon(Icons.clear));
+    await tester.pumpAndSettle();
+    final Material material = tester.widget<Material>(
+      find.descendant(
+        of: find.byType(BottomSheet),
+        matching: find.byType(Material),
+      ).first,
+    );
+    expect(material.elevation, modalElevation);
+  });
+
+  testWidgets('BottomSheet persistentElevation only sets elevation for persistent bottom sheet - Persistent', (WidgetTester tester) async {
+    const persistentElevation = 7.0;
+    const bottomSheetTheme = BottomSheetThemeData(
+      persistentElevation: persistentElevation,
+    );
+    await tester.pumpWidget(bottomSheetWithElevations(bottomSheetTheme));
+    await tester.tap(find.byIcon(Icons.check));
+    await tester.pumpAndSettle();
+    final Material material = tester.widget<Material>(
+      find.descendant(
+        of: find.byType(BottomSheet),
+        matching: find.byType(Material),
+      ).first,
+    );
+    expect(material.elevation, persistentElevation);
+  });
+
+  testWidgets('BottomSheet persistentElevation only sets elevation for persistent bottom sheet - Modal', (WidgetTester tester) async {
+    const persistentElevation = 7.0;
+    const bottomSheetTheme = BottomSheetThemeData(
+      persistentElevation: persistentElevation,
+    );
+    await tester.pumpWidget(bottomSheetWithElevations(bottomSheetTheme));
+    await tester.tap(find.byIcon(Icons.clear));
+    await tester.pumpAndSettle();
+    final Material material = tester.widget<Material>(
+      find.descendant(
+        of: find.byType(BottomSheet),
+        matching: find.byType(Material),
+      ).first,
+    );
+    expect(material.elevation, 0);
+  });
+}
+
+Widget bottomSheetWithElevations(BottomSheetThemeData bottomSheetTheme) {
+  return MaterialApp(
+    theme: ThemeData(bottomSheetTheme: bottomSheetTheme),
+    home: Scaffold(
+      body: Builder(
+        builder: (BuildContext context) {
+          return Column(
+              children: [
+                RawMaterialButton(
+                  child: const Icon(Icons.check),
```

Comment:
I think the tests would be a little less confusing if you put this as a Text widget with 'Show Persistent' and the other one as 'Show Modal' since it wasn't immediately clear to me that tapping check/clear do something

Reply:
Done